axelor-human-resource/src/main/java/com/axelor/apps/hr/service/lunch/voucher/LunchVoucherMgtServiceImpl.java

```diff
@@ -245,62 +245,46 @@ public int updateStock(
 
-    try (final Writer writer = new OutputStreamWriter(os)) {
+    File tempFile = MetaFiles.createTempFile(null, ".csv").toFile();
+    try (CSVPrinter printer = CSVFile.DEFAULT.withDelimiter(';').withQuoteAll().write(tempFile)) {
 
       List<String> header = new ArrayList<>();
-      header.add(escapeCsv(I18n.get("Company code")));
-      header.add(escapeCsv(I18n.get("Lunch Voucher's number")));
-      header.add(escapeCsv(I18n.get("Employee")));
-      header.add(escapeCsv(I18n.get("Lunch Voucher format")));
-
-      writer.write(Joiner.on(";").join(header));
+      header.add(I18n.get(ITranslation.EMPLOYEE_CODE));
+      header.add(I18n.get(ITranslation.EMPLOYEE_NAME_AND_SURNAME));
+      header.add(I18n.get(ITranslation.EMPLOYEE_LUNCH_VOUCHER_NUMBER));
+      printer.printRecord(header);
 
       for (LunchVoucherMgtLine lunchVoucherMgtLine : lunchVoucherMgt.getLunchVoucherMgtLineList()) {
 
         List<String> line = new ArrayList<>();
-        line.add(escapeCsv(lunchVoucherMgt.getCompany().getCode()));
-        line.add(escapeCsv(lunchVoucherMgtLine.getLunchVoucherNumber().toString()));
-        line.add(escapeCsv(lunchVoucherMgtLine.getEmployee().getName()));
-        line.add(
-            escapeCsv(lunchVoucherMgtLine.getEmployee().getLunchVoucherFormatSelect().toString()));
-
-        writer.write("\n");
-        writer.write(Joiner.on(";").join(line));
+        line.add(lunchVoucherMgtLine.getEmployee().getExportCode());
+        line.add(lunchVoucherMgtLine.getEmployee().getContactPartner().getSimpleFullName());
+        line.add(lunchVoucherMgtLine.getLunchVoucherNumber().toString());
+        printer.printRecord(line);
       }
-
-      Beans.get(MetaFiles.class).upload(tempFile.toFile(), metaFile);
-
-    } catch (Exception e) {
-      Throwables.propagate(e);
-    } finally {
-      Files.deleteIfExists(tempFile);
     }
-    /*
-     */
+
     // lunchVoucherMgt.setExported(true);
-    lunchVoucherMgt.setCsvFile(metaFile);
+    lunchVoucherMgt.setCsvFile(
+        uploadMetaFile(
+            I18n.get("LunchVoucherCommand")
+                + " - "
+                + appBaseService
+                    .getTodayDate(lunchVoucherMgt.getCompany())
+                    .format(DateTimeFormatter.ISO_DATE)
+                + ".csv",
+            tempFile));
     lunchVoucherMgt.setExportDateTime(
         appBaseService.getTodayDateTime(lunchVoucherMgt.getCompany()).toLocalDateTime());
-
     lunchVoucherMgtRepository.save(lunchVoucherMgt);
   }
 
-  protected String escapeCsv(String value) {
-    if (value == null) return "";
-    if (value.indexOf('"') > -1) value = value.replaceAll("\"", "\"\"");
-    return '"' + value + '"';
+  protected MetaFile uploadMetaFile(String fileName, File tempFile) throws IOException {
+    MetaFile metaFile = new MetaFile();
+    metaFile.setFileName(fileName);
+    metaFiles.upload(tempFile, metaFile);
+    return metaFile;
```

Comment:
This method is duplicated in LeaveServiceImpl, you could create a new service in axelor-base MetafileUploadService.